tests/test_perf_target_scale.py

```
@@ -0,0 +1,126 @@
+        self.assertEqual(len(decoded["runs"]), 4)
+
+    def test_short_cpu_sample_is_rejected(self) -> None:
+        run = metrics.synthetic_run(200)
+        run["cpu"]["duration_seconds"] = metrics.MIN_CPU_SECONDS - 1
+        codes = {failure["code"] for failure in metrics.evaluate_run(run)}
+        self.assertIn("cpu_duration", codes)
+
+
+class TargetScaleCliTests(unittest.TestCase):
+    def test_self_test_cli_passes(self) -> None:
+        self.assertEqual(runner.main(["--self-test", "--output", "/tmp/cmux-target-scale-self-test.json"]), 0)
```

Ruff (0.16.2)

[error] Probable insecure usage of temporary file or directory: "/tmp/cmux-target-scale-self-test.json"

(S108)